File: packages/Server/Domain/Models/Health.cs
namespace Solidarity.Domain.Models;

public class Health
{
	public HealthStatus Status { get; init; }
	public IEnumerable<HealthCheck> Checks { get; init; } = new List<HealthCheck>();
}

public class HealthCheck : Health
{
	public string Key { get; init; } = string.Empty;
}

File: packages/Client/application/status/Health.ts
import { Api } from 'application'

export const enum HealthCheckStatus { Unhealthy, Degraded, Healthy }

export class Health {
	static async get() {
		const health = await Api.get<Health>('/health', { noHttpErrorOnErrorStatusCode: true })
		return new Health(health)
	}

	constructor(init?: Partial<Health>) {
		Object.assign(this, init)
	}

	readonly status!: HealthCheckStatus
	readonly checks!: Array<HealthCheck>
}

export type HealthCheck = {
	readonly key: string
	readonly status: HealthCheckStatus
	readonly checks: Array<HealthCheck>
}